pkg/controller/metrics/metrics.go

```diff
@@ -225,6 +235,27 @@ func (mc *Calculator) Start(ctx context.Context) error {
 						mcLog.Infof("cleared metric: %v", metricClusterDeploymentSyncsetPaused)
 					}
 				}
+
+				// For a cluster resuming from hibernated state or trying to hibernate, we report the seconds for that
+				// transition time to the desired power state
+				hibernatingCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions,
+					hivev1.ClusterHibernatingCondition)
+				if hibernatingCondition != nil {
+					if hibernatingCondition.Reason == hivev1.ResumingHibernationReason ||
+						hibernatingCondition.Reason == hivev1.StoppingHibernationReason {
+						metricClusterDeploymentPowerStateTransitionSeconds.WithLabelValues(cd.Name,
+							string(cd.Spec.PowerState)).Set(time.Since(
+							hibernatingCondition.LastTransitionTime.Time).Seconds())
+					} else {
+						cleared := metricClusterDeploymentPowerStateTransitionSeconds.Delete(map[string]string{
+							"cluster_deployment": cd.Name,
+							"desired_powerstate": string(cd.Spec.PowerState),
+						})
+						if cleared {
+							mcLog.Infof("cleared metric: %v", metricClusterDeploymentPowerStateTransitionSeconds)
+						}
+					}
+				}
```

Comment:
I don't think a per cluster labelled metric is a good idea here, that should be a last resort as we've learned with the provisioning delay issues and the cardinality we hit. 

There are a few other ways that would be less intensive on prometheus.

(1) Use the pre-existing cluster condition counts with and an alert to watch for a timeout reason, and separate tooling to tell you which clusters are in that state for SRE.
(2) A new gauge metric for stuck resuming/hibernating clusters beyond a threshold, separate tooling to find you which clusters those are.
(2) A whole system histogram for reporting how long each successful transition took, allowing us to see the average across all operations. This of course would not have entries for stuck transitioning items that never make it, which is a big flaw.

I think the per cluster thing is an anti-pattern we should avoid if possible, if grafana keeps at least 2 weeks of results, this can be a whole lot of time series in a 1k cluster with a lot of hibernating resuming going on.

Reply:
> per cluster labelled

D'oh! Sorry, I totally missed that :(